### ISSUE TITLE
TCP retransmission after deleted participant [13710]

### ISSUE THREAD

**Issue by ilikecola** [eProsima/Fast-DDS#2318]
After delete a participant and recreate the participant, messages can be sent under UDP but not under TCP.

## Expected Behavior
messages can be sent under UDP and TCP after participant recreated.

## Current Behavior
messages can not be sent under TCP after participant recreated.

## Steps to Reproduce
1.start publisher and subscrber whose transport is TCP
2.delete publisher's participant and after 10s recreate publisher's participant

## System information
Fast-RTPS version: 2.4.0
OS:Ubuntu:20.04

## Additional context
TCP connection can be established after participant recreated.

## Example code

```
int msgsent = 0;
    char ch = 'y';
    do
    {
            writer_->write(&st);
            ++msgsent;
            std::cout << "Sending sample, count=" << msgsent << std::endl;
            std::this_thread::sleep_for(std::chrono::milliseconds(1000));
    } while (msgsent<10);
	participant_->delete_contained_entities();
	DomainParticipantFactory::get_instance()->delete_participant(participant_);
	std::cout << "participant deleted and wait 10s to rebuild... " << std::endl;
	std::this_thread::sleep_for(std::chrono::milliseconds(10000));
	init();
	do
	{
		writer_->write(&st);
		++msgsent;
		std::cout << "Sending sample, count=" << msgsent << std::endl;
		std::this_thread::sleep_for(std::chrono::milliseconds(1000));
	} while (true);
```

**Comment by JLBuenoLopez-eProsima** [comment]
@ilikecola,

This issue might have been fixed by #2470, which has been also backported to all supported Fast DDS releases. Would you mind checking against one of these releases if the issue is still happening?

**Comment by JLBuenoLopez-eProsima** [comment]
I am going to close this issue due to inactivity. A possible fix was merged a couple of months ago and no feedback from the user has been received. @ilikecola, please, feel free to reopen if the issue is still happening after applying changes in #2470.
